demos/build.gradle

```diff
@@ -20,7 +16,45 @@ subprojects {
   dependencies {
     implementation project(':ehcache-impl')
     implementation 'javax.servlet:javax.servlet-api:3.1.0'
-    runtimeOnly 'ch.qos.logback:logback-classic:1.2.3'
+    runtimeOnly 'ch.qos.logback:logback-classic:1.2.11'
     runtimeOnly 'com.h2database:h2:1.4.196'
   }
+
+  configurations.all {
```

Comment:
`appRun` fails if the SLF4J clause is omitted.

```plain
Execution failed for task ':demos:00-NoCache:appRun'.
> Could not resolve all dependencies for configuration ':demos:00-NoCache:runtimeClasspath'.
   > Conflict(s) found for the following module(s):
       - org.slf4j:slf4j-api between versions 1.7.36 and 1.7.32
     Run with:
         --scan or
         :demos:00-NoCache:dependencyInsight --configuration runtimeClasspath --dependency org.slf4j:slf4j-api
     to get more insight on how to solve the conflict.
```

Addressing this with an exclusion on `logback-classic`, i.e.

```groovy
    runtimeOnly ('ch.qos.logback:logback-classic:1.2.11') {
      exclude group: 'org.slf4j', module: 'slf4j-api'
    }
```

yields

```plain
Execution failed for task ':demos:00-NoCache:appRun'.
> Could not resolve all dependencies for configuration ':demos:00-NoCache:grettyRunnerJetty9'.
   > Conflict(s) found for the following module(s):
       - org.slf4j:slf4j-api between versions 1.7.32 and 1.7.7
     Run with:
         --scan or
         :demos:00-NoCache:dependencyInsight --configuration grettyRunnerJetty9 --dependency org.slf4j:slf4j-api
     to get more insight on how to solve the conflict.
```

because of the version dragged in by gretty.

Have you another solution to suggest?